### PR TITLE
Playlist import compares path case-insensitively

### PR DESCRIPTION
**Pull request by daladim**
When generated on a Windows machine (where the filesystem paths are case-insensitive), M3U files may contain paths that do not exactly match the filesystem names. E.g., the M3U file can reference `Music/John John/Song.mp3`, while the actual file is stored as `music/John john/song.MP3`

Until now, this prevented S2 to correctly import the playlist (as this file would be silently ignored).
This MR fixes this issue. Such files are now correctly imported.

### CODE REVIEW

File: androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt [hunk 1]
```diff
@@ -118,16 +123,17 @@ class TaglibMediaProvider(
...
                         sanitisedSongPaths.keys.firstOrNull { songPath ->
                             when {
-                                songPath == entry.location -> {
+                                songPath == lowercaseEntryLocation -> {
                                     true
                                 }
-                                songPath.length > entry.location.length -> {
-                                    songPath.contains(entry.location)
+                                songPath.length > lowercaseEntryLocation.length -> {
+                                    songPath.contains(lowercaseEntryLocation)
```
Comment (timusus) [inline androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt:133]:
```suggestion
                                    songPath.contains(other = entry.location, ignoreCase = true)
```

File: androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt [hunk 2]
```diff
@@ -118,16 +123,17 @@ class TaglibMediaProvider(
...
                                 }
                                 else -> {
-                                    entry.location.contains(songPath)
+                                    lowercaseEntryLocation.contains(songPath)
```
Comment (timusus) [inline androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt:136]:
```suggestion
                                    lowercaseEntryLocation.contains(other = songPath, ignoreCase = true)
```

File: androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt [hunk 3]
```diff
@@ -118,16 +123,17 @@ class TaglibMediaProvider(
                             )
                         )
 
+                        val lowercaseEntryLocation = entry.location.lowercase()
```
Comment (timusus) [inline androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt:126]:
```suggestion

```

File: androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt [hunk 4]
```diff
@@ -118,16 +123,17 @@ class TaglibMediaProvider(
...
                         sanitisedSongPaths.keys.firstOrNull { songPath ->
                             when {
-                                songPath == entry.location -> {
+                                songPath == lowercaseEntryLocation -> {
```
Comment (timusus) [inline androidApp/main/mediaprovider/local/src/main/java/com/simplecityapps/localmediaprovider/local/provider/taglib/TaglibMediaProvider.kt:129]:
```suggestion
                                songPath.equals(entry.location, ignoreCase = true) -> {
```

**Review by timusus — APPROVED** [review]
Here I've just suggested some changes which compare ignoring case in a slightly more idiomatic fashion. 

Bit of a nitpick, happy to approve either way :)

**Comment by daladim** [comment]
Thanks for these suggestions, this definitely looks better this way

**Review by timusus — APPROVED** [review]

**Comment by timusus** [comment]
@daladim sorry to kick this one back again, the tests seem to be failing. I'll have a look myself when I get a chance, but you might get to it quicker..

**Review by timusus — APPROVED** [review]

**Pull request merged**